### ISSUE TITLE
Not able to connect SuperSet with MySQL and Postgres Database

### ISSUE THREAD

**Issue by GaryMam** [apache/superset#15741]
!!!!URGENT!!!!

I have installed MYSQL and Postgres database on my machine but when I am trying to add these datasource on their default port it gives me below error : 

1. MySQL : "Port 3306 is closed". I even tried other ports but not working.
2. Postgres : "Port 5432 is closed". I even tried other ports but not working.

I ran Port Scan on my MAC to see what all port are open and below is the result : 

Port Scan has started…

Port Scanning host: 127.0.0.1

	 Open TCP Port: 	631    		ipp
	 Open TCP Port: 	3306   		mysql
	 Open TCP Port: 	5432   		postgresql
	 Open TCP Port: 	8088   		radan-http
	 Open TCP Port: 	33060
	 Open TCP Port: 	50589
Port Scan has completed…

Please help me connecting datasource to Superset. Quick response will be highly appreciated.

Warm Regards,
GaryMam

**Comment by betodealmeida** [comment]
Hi, @GaryMam. Can you give more details of how you're trying to connect? What are you specifying as the host?

**Comment by nytai** [comment]
If you're running in docker on mac, you might have to specify `docker.for.mac.localhost` instead of localhost/127.0.0.1 

https://forums.docker.com/t/understanding-the-docker-for-mac-localhost-behavior/41921

**Comment by GaryMam** [comment]
I have installed Mysql on my MAC and created successful connention in database. When I am trying to add Mysql database as a source in Superset with host as "localhost" and port is "3306", it gives me error that "Port 3306" is closed.

**Comment by srinify** [comment]
Hey @GaryMam if you join our Slack community, I can help you out there. To join the Slack, head to https://superset.apache.org/community and click the Slack link.

I'm closing this issue for now, as this isn't a Superset bug or feature request :]

**Comment by srkds** [comment]
> If you're running in docker on mac, you might have to specify `host.docker.internal` instead of localhost/127.0.0.1
> 
> https://forums.docker.com/t/understanding-the-docker-for-mac-localhost-behavior/41921

Thanks, @nytai 👍
I was facing a similar issue in windows but fixed it by replacing ` localhost/127.0.0.1` with `host.docker.internal`

**Comment by muthubuntu** [comment]
One more solution to try is to use the GATEWAY IP address from the container hosting postgresql:

`docker inspect container-name-postgresql`

Scroll to the bottom of the output to find the NETWORKS: 

```
 "Networks": {
                "bridge": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "DriverOpts": null,
                    "GwPriority": 0,
                    "NetworkID": "149df16a0211a7d18ab4311643db23db55fd5780a5e62903bd892d92389d191c",
                    "EndpointID": "b8fc1f0fff11ff4a4322808aa893d745e2de0cdef69b336b52231379aa516edf",
                    "Gateway": "172.17.0.1",
                    "IPAddress": "172.17.0.3",
                    "MacAddress": "6a:dc:01:9e:be:49",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "DNSNames": null
                }
            }
```
The GATEWAY is set to 172.17.0.1

When I use this as my HOST in the Superset > Connect to Postgresql it connects quickly and I can see the data in my connected postgresql database.
